mtcpclient/calculate.go

```diff
@@ -16,18 +14,16 @@ type metricsCollectionStats struct {
 	numberOfConnections int
 }
 
-func (gc GroupOfConnections) calculateMetricsReport(status tcpclient.ConnectionStatus) (mr metricsCollectionStats) {
+func (gc GroupOfConnections) calculateMetricsReport() (mr metricsCollectionStats) {
 	// TODO: There's something i don't like... initiatlizing values in the first loop, and the standard deviation
 	//  requiring an extra pass considering all items... i'd move initialization out of the loop, and maybe iterate
 	//  over a filtered list rather than several loops over the original one, and maybe use specific generic functions...
 	//  requires further thinking in any case
 	// TODO: (2) rather than calculate a metrics report with a filter on the connection status, lets just promote splitting
 	// group of connections and running these functions over the subsets...
 	mr.total = 0
-	mr.numberOfConnections = 0
-	for _, item := range gc {
-		if item.GetConnectionStatus() == status {
-			mr.numberOfConnections++
+	if mr.numberOfConnections = len(gc.connections); mr.numberOfConnections > 0 {
```

Comment:
i don't fully understand why you are protecting an iteration in case the slice is empty... an empty slice should just not enter into the loop, so i guess this is there in case gc.connections is not initialized as an slice (or something similar), in case the "len()" function is able not to raise an error in case of the slice being nil... so I'd try to drop...

Reply:
`panic: runtime error: integer divide by zero`, I'm protecting this: `mr.avg = mr.total / time.Duration(mr.numberOfConnections)`

Reply:
Ok. I'd suggest then an "early return" rather than encapsulating the whole function by an if...